### PR TITLE
Add tokenlist validation

### PR DESCRIPTION
**Pull request by unanoc**
closes https://github.com/trustwallet/assets-go-libs/issues/48

### CODE REVIEW

File: blockchains/smartchain/assets/0x0D8Ce2A99Bb6e3B7Db580eD848240e4a0F9aE153/info.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "Filecoin",
+    "name": "Binance-Peg Filecoin",
```
Comment (vikmeup) [inline blockchains/smartchain/assets/0x0D8Ce2A99Bb6e3B7Db580eD848240e4a0F9aE153/info.json:2]: 👍

File: internal/processor/validators.go [hunk 2]
```diff
@@ -352,6 +352,65 @@ func (s *Service) ValidateTokenListFile(f *file.AssetFile) error {
...
+	}
+
+	return nil
+}
+
+func compareTokenlistWithAssets(tokens []TokenItem, chain string) error {
+	compErr := validation.NewErrComposite()
+
+	for _, token := range tokens {
+		if token.Type == "coin" {
```
Comment (vikmeup) [inline internal/processor/validators.go:373]:
```suggestion
		if token.Type == types.Coin {
```

Comment (vikmeup) [reply internal/processor/validators.go:373]: https://github.com/trustwallet/go-primitives/blob/master/types/token.go#L27

File: internal/processor/validators.go [hunk 3]
```diff
@@ -352,6 +352,65 @@ func (s *Service) ValidateTokenListFile(f *file.AssetFile) error {
...
+
+		if token.Type != *infoAsset.Type {
+			compErr.Append(fmt.Errorf("field type differs from %s", assetPath))
+		}
+
+		if token.Symbol != *infoAsset.Symbol {
+			compErr.Append(fmt.Errorf("field symbol differs from %s", assetPath))
+		}
+
+		if token.Decimals != uint(*infoAsset.Decimals) {
```
Comment (vikmeup) [inline internal/processor/validators.go:405]: Add name validation

File: blockchains/ethereum/tokenlist.json [hunk 4]
```diff
@@ -3168,7 +3168,7 @@
             "type": "ERC20",
             "address": "0xdd974D5C2e2928deA5F71b9825b8b646686BD200",
             "name": "Kyber Network Crystal",
```
Comment (vikmeup) [inline blockchains/ethereum/tokenlist.json:3170]: @leedaniil can you delete `KNCL` ? because the token is already migrated to the new token.

Comment (vikmeup) [reply blockchains/ethereum/tokenlist.json:3170]: I think you should add a check to check status of the token, if not active - throw an error

Comment (leedaniil) [reply blockchains/ethereum/tokenlist.json:3170]: okay, I marked it as abandoned

**Comment by Brentus1** [comment]
0xd4bA6924be11e449F5c46b1744c6C5a75f492661

**Comment by Brentus1** [comment]
https://github.com/trustwallet/assets/pull/17111/files/c80ecb67404b72269ef1f73f7967dd5f31d37a5d#diff-f8fa098be02034abfb041eebf5f2eb759b2f1a9c3daf3ce12d6c952e42ea6e61

**Pull request merged**